onnxruntime/core/providers/cpu/ml/array_feature_extractor.cc

```diff
@@ -53,36 +53,43 @@ template <typename T>
 common::Status ArrayFeatureExtractorOp<T>::Compute(OpKernelContext* context) const {
   const Tensor& X = *context->Input<Tensor>(0);
   const TensorShape& x_shape = X.Shape();
-  const vector<int64_t>& x_dims = x_shape.GetDims();
+  const int64_t x_num_dims = x_shape.NumDimensions();
   const T* x_data = X.template Data<T>();
 
-  if (x_dims.empty()) {
+  if (x_num_dims == 0) {
     return Status(ONNXRUNTIME, INVALID_ARGUMENT, "Invalid argument: X input has empty dimensions.");
   }
 
-  int64_t stride = x_dims.size() == 1 ? x_dims[0] : x_dims[1];
-  int64_t N = x_dims.size() == 1 ? 1 : x_dims[0];
+  const int64_t stride = x_shape[x_num_dims - 1];
 
   const Tensor& Y = *context->Input<Tensor>(1);
   const TensorShape& y_shape = Y.Shape();
   const int64_t* y_data = Y.template Data<int64_t>();
-  int64_t num_indices = y_shape.Size();
+  const int64_t num_indices = y_shape.Size();
 
   // validate Y
   if (num_indices == 0) {
     return Status(ONNXRUNTIME, INVALID_ARGUMENT, "Invalid Y argument: num_indices = 0");
   }
 
-  if (num_indices - 1 >= stride) {
-    std::ostringstream err_msg;
-    err_msg << "Invalid Y argument: num_indices - 1 (" << num_indices - 1 << ") >= stride (" << stride << ")";
-    return Status(ONNXRUNTIME, INVALID_ARGUMENT, err_msg.str());
+  for (int64_t i = 0; i < num_indices; ++i) {
+    if (y_data[i] >= stride) {
+      return ONNXRUNTIME_MAKE_STATUS(
+          ONNXRUNTIME, INVALID_ARGUMENT,
+          "Invalid Y argument: index is out of range: Y[", i, "] (", y_data[i], ") >=", stride);
+    }
   }
 
-  Tensor* Z = context->Output(0, TensorShape({N, num_indices}));
+  const TensorShape z_shape = [num_indices, &x_shape]() {
```

Comment:
Why creating a lambda here?

Reply:
using a lambda to initialize a const variable so i don't have to make it non-const and then update it after creation